### PR TITLE
Network Hierarchies reference doc.

### PR DESCRIPTION
**Pull request by AMZN-Olex**
Signed-off-by: Olex Lozitskiy <5432499+AMZN-Olex@users.noreply.github.com>>

## Change summary

- Network Hierarchies reference doc page.
- Minor corrections in other network docs.

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+With the hierarchy debugger enabled, you will be able to browse hierarchies and get debug text in the world over the root entities of hierarchies.
+
+![Hierarchy Debug Overlay](/images/user-guide/gems/reference/multiplayer/imgui_hierarchy_debugger_overlay.png)
+
+Using this debugger, you can find the following information about a hierarchy:
+- Its root entity name
+- Total number of entities in the hierarchy
+- Entity name of each member within the hierarchy
+- `AZ::EntityId` of each member
+- Network id of each member, `Multiplayer::NetEntityId`
```
Comment (sptramer) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md:70]:
```suggestion
- The `AZ::EntityId` of each member
- The network ID of each member, `Multiplayer::NetEntityId`
```

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+- Its hierarchical role within the hierarchy, either child, root, or an inner root that acts as a child in the case of nested hierarchies.
+
+
+### Adding Input Processing to Network Hierarchies
+
+Any entity that has component(s) that perform multiplayer input processing component need to have either:
+- `Local Prediction Input Component`
+- `Network Hierarchy Root Component`
+- `Network Hierarchy Child Component`
```
Comment (sptramer) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md:79]:
```suggestion
Any entity that has component(s) that perform multiplayer input processing component needs to have one of the following components:
- **Local Prediction Input**
- **Network Hierarchy Root**
- **Network Hierarchy Child**
```

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+
+    <NetworkInput Type="float" Name="InputValue" Init="0.0f" />
+
+</Component>
+```
+
+Add a component to provide multiplayer input processing:
+
+![Hierarchy Debug Overlay](/images/user-guide/gems/reference/multiplayer/hierarchy_child_entity_with_input_processing_component.png)
```
Comment (sptramer) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md:100]:
```suggestion
```

Redundant, can be removed.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+As a general rule, a hierarchy must have one `Local Prediction Input Component` at its top parent entity to process multiplayer input.
+
+![Hierarchy Debug Overlay](/images/user-guide/gems/reference/multiplayer/hierarchy_root_with_local_prediction_component.png)
+
+While hierarchical child entities need only to have `Network Hierarchy Child Component` to process their multiplayer input.
+
+![Hierarchy Debug Overlay](/images/user-guide/gems/reference/multiplayer/hierarchy_child_entity_with_input_processing.png)
```
Comment (sptramer) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md:108]:
```suggestion
As a general rule, a hierarchy must have one **Local Prediction Input** component at its top parent entity to process multiplayer input, while hierarchical child entities need only to have a **Network Hierarchy Child** component to process their multiplayer input.
```

Remove unnecessary images.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+
+
+### Creating Network Hierarchies at Runtime
+
+Network Hierarchies are automatically updated, created and disbanded at runtime whenever parent-child relationship changes among entities with `Network Hierarchy Root Component` or `Network Hierarchy Child Component`.
+
+In other words, you can use `AZ::TransformBus` to attach and detach entities to other entities. If such entities have hierarchical components, any hierarchies involved will be automatically rebuilt at runtime.
+
```
Comment (sptramer) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md:117]:
```suggestion
Network hierarchies are automatically updated, created and disbanded at runtime whenever parent-child relationship changes among entities with either a hierarchy root or hierarchy child component. When the `AZ::TransformBus` EBus is used to change the attachments of entities, the hierarchies are updated at runtime.
```

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+### Hierarchies of Hierarchies
+
+It is possible to create a hierarchy out of several hierarchies by attaching the root of one hierarchy to an entity from another hierarchy. In such cases, an inner root (an entity with `Network Hierarchy Root Component`) becomes a hierarchical child until it is detached from the hierarchy. This is useful when constructing vehicles, weapons and players out of different prefabs that are attached and detached at runtime as needed.
+
+
+{{<note>}}
+You can find a lot of extra behind the scenes information by browsing the unit tests for network hierarchies here:
+https://github.com/o3de/o3de/blob/development/Gems/Multiplayer/Code/Tests/ServerHierarchyTests.cpp
```
Comment (sptramer) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md:125]:
```suggestion
For more information on network hierarchies, read the [ServerHierarchyTests source](https://github.com/o3de/o3de/blob/development/Gems/Multiplayer/Code/Tests/ServerHierarchyTests.cpp).
```

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+### Creating Network Hierarchies at Runtime
+
+Network Hierarchies are automatically updated, created and disbanded at runtime whenever parent-child relationship changes among entities with `Network Hierarchy Root Component` or `Network Hierarchy Child Component`.
+
+In other words, you can use `AZ::TransformBus` to attach and detach entities to other entities. If such entities have hierarchical components, any hierarchies involved will be automatically rebuilt at runtime.
+
+### Hierarchies of Hierarchies
+
+It is possible to create a hierarchy out of several hierarchies by attaching the root of one hierarchy to an entity from another hierarchy. In such cases, an inner root (an entity with `Network Hierarchy Root Component`) becomes a hierarchical child until it is detached from the hierarchy. This is useful when constructing vehicles, weapons and players out of different prefabs that are attached and detached at runtime as needed.
```
Comment (sptramer) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md:120]:
```suggestion
It is possible to create a hierarchy out of several hierarchies by attaching the root of one hierarchy to an entity from another hierarchy. In such cases, an inner root component becomes a hierarchical child of this new parent, until it is detached from the hierarchy. This is useful when constructing vehicles, weapons and players out of different prefabs that are attached and detached at runtime as needed.
```

**Review by sptramer — CHANGES_REQUESTED** [review]
Some style and wording changes, and request to remove unneeded images.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    - Network Transform Component
+
+
+1. Add a Mesh component for visual information in the Editor.
+
+1. Here is an example of a simple network entity:
+
+    ![Starting Network Entity](/images/user-guide/gems/reference/multiplayer/starting_network_entity.png)
+
+1. In order to make it a hierarchical root entity, add `Network Hierarchy Root Component`:
```
[the remark this answers is not in the document]
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+    ![Starting Hierarchy Root Entity](/images/user-guide/gems/reference/multiplayer/starting_hierarchy_root_entity.png)
+
+1. This entity will be a parent entity for hierarchical child entities.
+
+1. A hierarchical child entity is formed by adding `Network Hierarchy Child Component`.
```
[the remark this answers is not in the document]
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+
+1. Here is an example of a simple network entity:
+
+    ![Starting Network Entity](/images/user-guide/gems/reference/multiplayer/starting_network_entity.png)
+
+1. In order to make it a hierarchical root entity, add `Network Hierarchy Root Component`:
+
+    ![Starting Hierarchy Root Entity](/images/user-guide/gems/reference/multiplayer/starting_hierarchy_root_entity.png)
+
+1. This entity will be a parent entity for hierarchical child entities.
```
[the remark this answers is not in the document]
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+
+1. A hierarchical child entity is formed by adding `Network Hierarchy Child Component`.
+
+    ![Starting Hierarchy Child Entity](/images/user-guide/gems/reference/multiplayer/starting_hierarchy_child_entity.png)
+
+1. Notice that this entity is a child entity to the root entity.
+
+    ![Simple Hierarchy Example](/images/user-guide/gems/reference/multiplayer/simple_hierarchy.png)
```
[the remark this answers is not in the document]
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+
+1. Often a hierarchy is a group of entities that you want to spawn at runtime. In such cases you want to create a prefab out of the entities.
```
[the remark this answers is not in the document]
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+
+    ![Simple Hierarchy Prefab](/images/user-guide/gems/reference/multiplayer/simple_hierarchy_prefab.png)
+
+1. It is important that the entity with `Network Hierarchy Root Component` is at the top parent entity within the prefab. All other entities should have `Network Hierarchy Child Component` on them.
```
[the remark this answers is not in the document]
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+
+
+### ImGui Hierarchy Debugger
+
+O3DE Multiplayer Gem comes with a debugger to help you trouble shoot and analyze hierarchy structure in your level. It can be enabled by bringing up ImGui menu ('HOME' key by default), and then selecting `Multiplayer` -> `Multiplayer Hierarchy Debugger`.
```
[the remark this answers is not in the document]
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+
+![Turning On Hierarchy ImGui](/images/user-guide/gems/reference/multiplayer/turning_on_imgui_hierarchy_debugger.png)
+
```
[the remark this answers is not in the document]
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+With the hierarchy debugger enabled, you will be able to browse hierarchies and get debug text in the world over the root entities of hierarchies.
+
+![Hierarchy Debug Overlay](/images/user-guide/gems/reference/multiplayer/imgui_hierarchy_debugger_overlay.png)
+
```
[the remark this answers is not in the document]
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md (hunk 1 above)
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md (hunk 2 above)
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md (hunk 3 above)
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md:94]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md (hunk 4 above)
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md (hunk 5 above)
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md (hunk 7 above)
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md (hunk 6 above)
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Done.

**Comment by AMZN-Olex** [comment]
I've addressed PR feedback and updated the PR.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 17]
```diff
@@ -0,0 +1,110 @@
...
+    ![Starting Hierarchy Root Entity](/images/user-guide/gems/reference/multiplayer/starting_hierarchy_root_entity.png)
+
+This entity will be a parent entity for hierarchical child entities.
+
+5. A hierarchical child entity is formed by adding a **Network Hierarchy Child** component.
+
+    ![Starting Hierarchy Child Entity](/images/user-guide/gems/reference/multiplayer/starting_hierarchy_child_entity.png)
+
+Notice that this entity is a child entity to the root entity.
```
Comment (sptramer) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md:44]:
```suggestion
1. In order to make it a hierarchical root entity, add a **Network Hierarchy Root** component:

    ![Starting Hierarchy Root Entity](/images/user-guide/gems/reference/multiplayer/starting_hierarchy_root_entity.png)

    This entity will be a parent entity for hierarchical child entities.

1. A hierarchical child entity is formed by adding a **Network Hierarchy Child** component.

    ![Starting Hierarchy Child Entity](/images/user-guide/gems/reference/multiplayer/starting_hierarchy_child_entity.png)

    Notice that this entity is a child entity to the root entity.
```

Markdown indentation for lists is extremely whitespace-sensitive. See [Markdown Cheatsheet](https://www.markdownguide.org/basic-syntax/#ordered-lists). If you have to manually label a step in a list, it means that you tried to place text inline with a step and used the wrong amount of whitespace.

**Review by sptramer — CHANGES_REQUESTED** [review]
One minor change to fix some formatting issues.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md (hunk 17 above)
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Corrected whitespace, thanks for the doc link!

**Review by lmbr-pip — APPROVED** [review]

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+    Often a hierarchy is a group of entities that you want to spawn at runtime. In such cases you want to create a prefab out of the entities.
+
+    ![Simple Hierarchy Prefab](/images/user-guide/gems/reference/multiplayer/simple_hierarchy_prefab.png)
+
+    It is important that the entity with a **Network Hierarchy Root** component is at the top parent entity within the prefab. All other entities should have a **Network Hierarchy Child** component on them.
+
+
+### ImGui Hierarchy Debugger
+
+O3DE Multiplayer Gem comes with a debugger to help you trouble shoot and analyze hierarchy structure in your level. It can be enabled by bringing up ImGui menu (**HOME** key by default), and then selecting **Multiplayer** > **Multiplayer Hierarchy Debugger**. With the hierarchy debugger enabled, you will be able to browse hierarchies and get debug text in the world over the root entities of hierarchies.
```
[the remark this answers is not in the document]
Comment (AMZN-Olex) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/hierarchy.md]: Thanks, modified.

**Review by sptramer — APPROVED** [review]

**Pull request merged**